### PR TITLE
[ios][android] Update react-native-gesture-handler to 2.7.0

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

Upgrading the gesture handler for SDK 47
Closes ENG-6529

# How

- Migrated `react-native-gesture-handler` to the new vendoring mechanism
- Manually removed [`lib/build.gradle`](https://github.com/software-mansion/react-native-gesture-handler/blob/main/android/lib/build.gradle) from the source code so our vendored autolinking doesn't detect this as a separate library. This build.gradle should and will get removed from the repo (it's not used and uses pretty old versions that don't compile)
- `et uvm -m react-native-gesture-handler -c 2.7.0`

# Test Plan

NCL examples seem to work fine

### CODE REVIEW

File: android/vendored/unversioned/react-native-gesture-handler/android/build.gradle [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        versionCode 1
+        versionName "1.0"
+        buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
+        if (isNewArchitectureEnabled()) {
+            var appProject = rootProject.allprojects.find {it.plugins.hasPlugin('com.android.application')}
+            externalNativeBuild {
+                cmake {
+                    cppFlags "-O2 -frtti -fexceptions -Wall -Wno-unused-variable -fstack-protector-all"
+                    arguments "-DAPP_BUILD_DIR=${appProject.buildDir}",
+                        "-DREACT_NATIVE_DIR=${REACT_NATIVE_DIR}",
```
Comment (Kudo) [inline android/vendored/unversioned/react-native-gesture-handler/android/build.gradle:136]: oh no! gesture-handler has cpp code now, even though it's for fabric only 🤦‍♂️
that would increase the complexity in versioning the vendored module.
i'm just taking a note that i should aware of this 😅

File: apps/bare-expo/ios/Podfile.lock [hunk 2]
```diff
@@ -714,7 +714,7 @@ PODS:
     - React-Core
   - RNDateTimePicker (6.2.0):
     - React-Core
-  - RNGestureHandler (2.5.0):
+  - RNGestureHandler (2.6.2):
```
Comment (Kudo) [inline apps/bare-expo/ios/Podfile.lock:717]: the Podfile.lock in bare-expo should also update to 2.7.0.

File: tools/src/vendoring/config/expoGoConfig.ts [hunk 3]
```diff
@@ -58,6 +58,7 @@ const config: VendoringTargetConfig = {
       source: 'https://github.com/software-mansion/react-native-gesture-handler.git',
       semverPrefix: '~',
       ios: {},
+      android: {},
```
Comment (Kudo) [inline tools/src/vendoring/config/expoGoConfig.ts:61]: would you like to remove the vendoring code in _legacy.ts_ ?

File: yarn.lock [hunk 4]
```diff
@@ -17294,10 +17294,10 @@ react-native-fade-in-image@^1.6.1:
     react-mixin "^3.0.5"
     react-timer-mixin "^0.13.3"
 
-react-native-gesture-handler@~2.5.0:
-  version "2.5.0"
-  resolved "https://registry.yarnpkg.com/react-native-gesture-handler/-/react-native-gesture-handler-2.5.0.tgz#61385583570ed0a45a9ed142425e35f8fe8274fb"
-  integrity sha512-djZdcprFf08PZC332D+AeG5wcGeAPhzfCJtB3otUgOgTlvjVXmg/SLFdPJSpzLBqkRAmrC77tM79QgKbuLxkfw==
+react-native-gesture-handler@~2.6.2:
+  version "2.6.2"
```
Comment (Kudo) [inline yarn.lock:17298]: yarn.lock should also be updated to 2.7.0

**Review by Kudo — APPROVED** [review]
most changes look good to me. just as mentioned i should follow up with versioning later.
there're some nit comments for out-of-sync lock files. you could also update them after the pr landed.

thanks for migrating this module as new style vendoring 🔥

File: tools/src/vendoring/config/expoGoConfig.ts (hunk 3 above)
Comment (tsapeta) [reply tools/src/vendoring/config/expoGoConfig.ts:61]: I planned to do that, but there are more libs to delete from there, so I'll do that separately

**Pull request merged**